ds3-autogen-c/src/main/resources/templates/request-templates/AmazonS3GetRequest.ftl

```diff
@@ -0,0 +1,29 @@
+<#-- ********************************** -->
+<#-- Generate "AmazonS3 Get Request" -->
+<#--   Input: Source object             -->
+<#-- ********************************** -->
+<#list getRequests() as requestEntry>
+    <#if (requestEntry.getClassification().toString() == "spectras3") && (requestEntry.getVerb().toString() == "GET")>
```

Comment:
At the very least I think an improvement would be to use the same code generator, and then call out to a specific code generator as needed, that way the general logic is the same, but the parts that have to be unique can be.